### DISCUSSION TITLE
using own git repo as dependencies?

### DISCUSSION THREAD

**Discussion by geofarmbuilder** [integer32llc/rust-playground#1058] [Q&A] [answered]
Trying to add a personal repo as dependency, anyone has done it before? Modified the compiler/base/Cargo.toml. it wasn't picked up. any ideas? thanks

**Comment by shepmaster** [comment]
If you are building your own playground, then once you've modified the `Cargo.toml` you'll need to [rebuild the Docker containers](https://github.com/rust-lang/rust-playground/blob/main/compiler/build.sh).

If you mean you'd want to use your own git repo in an officially hosted playground, that will not happen for security reasons.

**Comment by geofarmbuilder** [comment]
Thanks for the quick response. I do host my own playground with some
modifications to the compiler scripts, trying to include my own repo in
Cargo.toml, couldn’t picked up while docker running. Anything I’m missing
in addition to add an entry to Cargo.toml?

**Reply by shepmaster** [reply]
What happens when you [rebuild the containers](https://github.com/rust-lang/rust-playground/blob/main/compiler/build.sh)? Is there an error?

**Comment by geofarmbuilder** [comment]
Rebuilding is fine, the problem I have is trying to import the one added in
editor, compiling throws error that crates not included.

**Reply by shepmaster** [reply]
When trying to get help here (and basically any other place on the internet), you will need to provide detailed information such as exact steps you took, exact changes made, exact error messages encountered, etc. I'll check back here in a few days/weeks and see if any have been provided.

**Reply by geofarmbuilder** [reply]
You're right. I was asking to see anyone had similar tries before. Here is what I've done. 
1: Goal is to re-use playground to compile some sample code with our own crate which haven't been published to crates.io. 
2 Steps done: 
2.1 modified compiler/base/Cargo.toml 
2.2 build the docker image with the modification 
2.3 restart playground service in background. 
2.4 write code with the crate(host on GitHub) imported, and then errors thown out indicates the crate hasn't been included 
3. Modifications 
diff --git a/compiler/base/Cargo.toml b/compiler/base/Cargo.toml
index f594048..17f9a1d 100644
--- a/compiler/base/Cargo.toml
+++ b/compiler/base/Cargo.toml
@@ -20,6 +20,16 @@ incremental = false
 codegen-units = 1
 debug = false
 
+[dependencies.zkwasm-rust-sdk]
+package = "zkwasm-rust-sdk"
+git = "https://github.com/DelphinusLab/zkWasm-rust.git"
+branch = "main"
+

diff --git a/compiler/base/crate-information.json b/compiler/base/crate-information.json
index 1052b0b..259c0a9 100644
--- a/compiler/base/crate-information.json
+++ b/compiler/base/crate-information.json
@@ -1723,5 +1723,12 @@
     "name": "zune-jpeg",
     "version": "0.4.11",
     "id": "zune_jpeg"
+  },
+  {
+    "name": "zkwasm-rust-sdk",
+    "git": "https://github.com/DelphinusLab/zkWasm-rust.git",
+    "branch": "main",
+    "id": "zkwasm_rust_sdk"

**Reply by geofarmbuilder** [reply, marked as answer]
figured out that dependencies will be removed when target set to wasm.
